src/NexusMods.Abstractions.Downloads/DownloadInfo.cs

```diff
@@ -0,0 +1,83 @@
+using System.Reactive.Disposables;
+using System.Runtime.CompilerServices;
+using DynamicData.Kernel;
+using JetBrains.Annotations;
+using NexusMods.Abstractions.Jobs;
+using NexusMods.Abstractions.Library.Models;
+using NexusMods.Abstractions.NexusWebApi.Types.V2;
+using NexusMods.Paths;
+using ReactiveUI;
+using ReactiveUI.Fody.Helpers;
+
+[assembly: InternalsVisibleTo("NexusMods.Library")]
+
+namespace NexusMods.Abstractions.Downloads;
+
+/// <summary>
+/// Represents download information for UI display.
+/// Thread safety is not guaranteed - UI consumers should use OnUI() or similar.
+/// </summary>
+[PublicAPI]
+public class DownloadInfo : ReactiveObject
+{
+    /// <summary>
+    /// Unique identifier for this download.
+    /// </summary>
+    public required DownloadId Id { get; init; }
+    
+    /// <summary>
+    /// The job ID for this download. Only valid for active downloads.
+    /// Consumers should generally use Id instead, but this is available for service implementations.
+    /// </summary>
+    public Optional<JobId> JobId { get; set; }
+    
+    /// <summary>
+    /// Display name of the download (typically mod name).
+    /// </summary>
+    [Reactive] public string Name { get; set; } = string.Empty;
```

Comment:
What about the original FileName of the download, is that something we would have access to or need access to?

`Cool mod.zip` I mean.

Reply:
This is set by `private string ExtractName(NexusModsDownloadJob nexusJob)` which uses the actual file name as fallback if unavailable. That code supports nexus and non-nexus downloads, but we're limiting to nexus downloads per last week discussion.